### ISSUE TITLE
embed survey widget

### ISSUE THREAD

**Issue by balmas** [alpheios-project/alpheios_nemo_ui#165]
I think we'll use survey sparrow to request feedback on the beta. We can embed a widget for the survey directly in the page.

**Comment by irina060981** [comment]
@balmas, @kirlat, @monzug  , I will place my thoughts/feedback here:

**1. Appearing the survey button**
Fom my point of view it is not annoying on the desktop and on the mobile, but on the mobile it could take rather a big place like this
![image](https://user-images.githubusercontent.com/12377640/58017949-d864fb80-7b44-11e9-9f27-bd4002f700c9.png)
And it has very the same background as texts so it is rather difficult to identify it correctly and to guess how to close it, if I am not going to answer it right now.

And the second point - I have suddenly close it after the first question and didn't find any way to open it again without reloading a page.

**Suggestion**
I am not very familiar with sparrow survey, but it would be useful to have 4 states of this widget:
   - active and welcome to survey (as it is now)
   - hidden after the survey is finished (it would be good not to show it again for this device/session)
   - opened during the survey is being answered
   - postponed (when the survey was close without all questions to be answered) - it could be a small icon at the same place - it won't take much place - but it would be easy to return to this

**2. Answering the questions** 
**Suggestions**
I like when the survey warns me - how much question is going to appear here.
Also if I understood the questions right - they are all about mobile version - so if I am not using mobile version - should I answer them (I was a little confused at the start because of it)
So may be if we are interested with the feeadback from mobile users - we should warn about it?

1 question - Which view is most similar to what you saw when reading texts on your mobile device?
      About screenshots - I think it would be better to give them order number , from my point of view.

2 question - How easy was it to select a word from the page for lookup on your mobile device?
      I think that 1-10 is too much variations, there are ussually used from 0-4 in tests 
      (completely incomfortable (0), not very easy(1), normal(2), easy but with some delays (3), not even mentioned  any problems(4)) I couldn't imagine more variants if make them as a description.

3 question - Did you use the application in portrait or landscape mode on your mobile device?
4 question - Did you use the application in portrait or landscape mode on your mobile device?
      These questions are duplicated

5 question - Which view looks most like one you saw on your mobile device after selecting a word?
6 question - Did you try using the inflection tables on the mobile device?

7 question - What text or texts would you most like to see added to the mobile reader?
      I wasn't able to find where should I type the answer.
      Only after several times I was able to notice the very light placeholder
      May be we could add to the question -type at the following line?
![image](https://user-images.githubusercontent.com/12377640/58020988-16661d80-7b4d-11e9-8409-45149ebfcf7a.png)

8 question - Did you need to zoom in or out of the Alpheios panel?

Answering your questions:
    > Do you like how this survey appear in general? Is it annoying or not? Is it too long or too short?

    Yes, I like the colors, that it is not very long and it is not very annoying (for the first reloads)
    
     >Are question presented well?

     It was a little confusing for me - do you interested only mobile feedback or not

      > Are questions easy to answer (from a technical standpoint)? How about questions with pictures? 
Are pictures clear enough to illustrate the question? Do you think the pictures are maybe too small to be easily understood as they are now?

      I really liked the idea with screenshots questions - although there are small (screenshots). Some questions.

      > Is there anything we can do to make this survey more user friendly?
      
      I think, that we could change the order of questions a little and this (for example as an idea):
      1. Do you use reader on a mobile device? (Yes, No) _If no, may be we should finish the survey or go to the last question, because all other quesations are about mobile device_
      2.  Which view is most similar to what you saw when reading texts on your device? (choice)
      3.  If you use it on a mobile device, did you use the application in portrait or landscape mode? (portrait, landscape, both, skip - I use on desktop)
       4. If you use it on a mobile device, how easy was it to select a word from the page for lookup? (skip - I use on desktop, choice from 0-4 with descriptions)
       5. Did you open inflection tables tab on mobile device? Did you like it? (I saw it only on desktop, I saw it on mobile and like it, I saw it on mobile and dislike it, I didn't see it)
        6. Which view looks most like one you saw on device after selecting a word?
        7. Did you need to zoom in or out of the Alpheios panel if you use it on mobile device?
        8. We would like to add more texts to our reader - What text or texts would you most like to see?

 > What would be the best time to show this survey to users: right after user enters the page, after some time, when user leaves the page?

        I think that it appears correctly now - after some time the user starts to read the page

>Monica was worried about it adding more to the load time of the page
       
        As I was testing it locally I didn't see any time delays or problems with it, but in production they could happen

> we can do the survey without embedding it, i.e. we could just put a link or button to open it
but the embedded popup seems to me like it might have the potential to generate more feedback . But it could also just annoy people.

         May be it would be useful to have both? To have a short survey embedded and the last point would be something like that:
          We would be very happy to get more feedback by the link :)
          It would be interesting for me for example, to have feedback about:
              where users use lookup (double-click, lookup in popup, panel or toolbar)
              Do they use settings (colors, fonts, dictionaries)?
              How do they open tabs from popup, toolbar or panel?
              Do they need so much notifications about login
              Do they like filters ui for word usage examples, wordlist
          
          May be someone would be ready to give some interesting free thoughts about features and UI

Hope my feedback was helpfull, if I have missed some questions, I am sorry. 
Finally, I like the idea with survey very much - it is the most exiciting and nervous part to get feedback from users :)

**Comment by monzug** [comment]
Irina, I think the questions are sample survey questions. on my mobile, I could not see the screenshots and could not zoom in either.

**Comment by balmas** [comment]
ok, the survey questions themselves still need to be figured out, but for now the widget is embedded.
It should display on the texts page until you close it.

Once you have closed it, it will reappear in the following situations:
1) first text page viewed after login
2) first text page viewed after logout
3) first text page after you clear your browser cookies

**Comment by monzug** [comment]
tested in text env build 1.0.0.16, Firefox, Chrome, iphone xr (Safari) and Samsung 5 mini (Chrome) worked as described above.
